### PR TITLE
Update doc matchers json

### PR DESCRIPTION
**Pull request by jesuswasrasta**
I verified the behavior running existing tests, then wrote the documentation.
In general, it would be nice having some examples for every assertion, too. What do you think?

### CODE REVIEW

**Comment by sksamuel** [comment]
Yes examples is a great idea. Maybe the matches.md file can link through to
more detailed examples

File: kotlintest-core/src/main/kotlin/io/kotlintest/Project.kt [hunk 1]
```diff
@@ -54,7 +54,7 @@ object Project {
   private val _projectlisteners = mutableListOf<ProjectListener>()
   private val _filters = mutableListOf<ProjectLevelFilter>()
   private var _specExecutionOrder: SpecExecutionOrder = LexicographicSpecExecutionOrder
-  private var writeSpecFailureFile: Boolean = true
+  private var writeSpecFailureFile: Boolean = false
```
Comment (LeoColman) [inline kotlintest-core/src/main/kotlin/io/kotlintest/Project.kt:57]: I'm not sure what this has to do with the feature

Comment (sksamuel) [reply kotlintest-core/src/main/kotlin/io/kotlintest/Project.kt:57]: That's on master too.

**Comment by sksamuel** [comment]
great PR thanks

**Pull request merged**